felix/fv/workload/workload.go

```diff
@@ -94,13 +95,8 @@ func (w *Workload) Stop() {
 		log.Info("Stop no-op because nil workload")
 	} else {
 		log.WithField("workload", w).Info("Stop")
-		output, err := w.C.ExecOutput("cat", fmt.Sprintf("/tmp/%v", w.Name))
-		Expect(err).NotTo(HaveOccurred(), "failed to run docker exec command to get workload pid")
-		pid := strings.TrimSpace(output)
-		w.C.Exec("kill", pid)
-		_ = w.C.ExecMayFail("ip", "link", "del", w.InterfaceName)
-		_ = w.C.ExecMayFail("ip", "netns", "del", w.NamespaceID())
-		_, err = w.runCmd.Process.Wait()
+		_ = w.C.ExecMayFail("sh", "-c", fmt.Sprintf("kill -9 %s & ip link del %s & ip netns del %s & wait", w.pid, w.InterfaceName, w.NamespaceID()))
```

Comment:
Hmm, there are a few details changing here, are you sure they are all intended and reliable?
- If `kill` failed, we'd previously fail the test at that point.
- Added `-9` to the kill.  Are you sure we didn't want to give the process a chance to shut down gracefully?
- `kill` and `ip` commands happening in parallel, whereas previously they were sequenced.  Will the namespace deletion fail if the kernel sees it before link deletion?

Reply:
Now that I've switched to `kill -9`, I think it can only fail is the process isn't running, which is probably OK given this is only called during AfterEach shut-down.  I changed to that to reduce the output logged on shutdown.  IIRC it was emitting a panic trace before, which was misleading/wasteful.

The other two commands seem OK to run in either order.  Deleting the netns actually deletes the veth too if the workload side of the veth was correctly moved into the netns but I thought it best to assume that moving the veth failed.

Reply:
Fair enough; make sense, and I'm happy if you think this will all be fine.

Reply:
I decided that I wasn't happy that there was no timeout on the Wait() call, especially if I increase a little extra risk here.  Added one along with the other markups.